.github/workflows/update_dependencies.yml

```diff
@@ -0,0 +1,68 @@
+name: Update dependencies
+on:
+  schedule:
+    # Run automatically at 6AM PST Tuesday
+    - cron: '0 6 * * 2'
+  workflow_dispatch:
+
+jobs:
+  update_dependencies_and_test:
+    name: Update Dependencies & Test
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    strategy:
+      matrix:
+        java: [ 8, 11, 17 ]
+
+    steps:
+      - name: Checkout java-http-client
+        uses: actions/checkout@v2
+
+      - name: Updating semver dependencies
+        run: make update-deps
+
+      - name: Set up Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.java }}
+          cache: 'maven'
+
+      - run: mvn install -Dgpg.skip -Dmaven.javadoc.skip=true -B -V
+
+  commit_dependencies:
+    name: Commit Updated Dependencies
+    if: success()
+    needs: [ update_dependencies_and_test ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout java-http-client
+        uses: actions/checkout@v2
+
+      - name: Updating semver dependencies
+        run: make update-deps
+
+      - name: Add & Commit
+        uses: EndBug/add-and-commit@v8.0.2
+        with:
+          add: 'pom.xml'
+          default_author: 'github_actions'
+          message: 'Chore: update sendgrid-java dependencies'
```

Comment:
nit: Think we use lowercase `chore` normally.

Also, says `sendgrid-java` instead of `java-http-client`